pkg/compose/attach.go

```diff
@@ -105,12 +106,18 @@ func (s *composeService) attachContainerStreams(ctx context.Context, container s
 	}()
 
 	if in != nil && stdin != nil {
-		err := in.SetRawTerminal()
+		state, err := term.SetRawTerminal(in.FD())
 		if err != nil {
-			return restore, err
+			return restore, detached, err
+		}
+		restore = func() {
+			term.RestoreTerminal(in.FD(), state)
 		}
 		go func() {
-			io.Copy(stdin, in) //nolint:errcheck
+			_, err := io.Copy(stdin, r) //nolint:errcheck
+			if _, ok := err.(term.EscapeError); ok {
+				detached <- true
```

Comment:
Can it be detached several times? Otherwise it could be
```suggestion
				close(detached)
```
?

Reply:
I expect we can throw away all this code to just adopt docker/cli run/exec commands